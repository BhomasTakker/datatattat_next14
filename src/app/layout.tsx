const DummyPlayer = () => {
	// console.log("DummyPlayer mounted");
	return null;
};

export default async function RootLayout({
	children,
}: Readonly<{
	children: React.ReactNode;
}>) {
	return (
		<div>
			{children}
			<DummyPlayer />
		</div>
	);
}
