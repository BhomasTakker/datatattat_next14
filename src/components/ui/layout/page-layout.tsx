import { MainFooter } from "@/components/footer/main-footer";
import styles from "./page-layout.module.scss";
import { geistMono, geistSans } from "@/fonts";
import { initialiseServices } from "@/lib/services/intialise-services";
import { Providers } from "@/components/providers/Providers";
import "../../../scss/globals.scss";

export const PageLayout = ({ children }: { children: React.ReactNode }) => {
	initialiseServices();
	return (
		<html lang="en">
			<Providers>
				<body
					className={`${geistSans.variable} ${geistMono.variable} ${styles.root}`}
				>
					{children}
					<MainFooter />
				</body>
			</Providers>
		</html>
	);
};
