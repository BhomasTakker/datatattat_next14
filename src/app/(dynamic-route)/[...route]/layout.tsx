import type { Metadata } from "next";
import { MainHeader } from "../../../components/header/main-header";
import { PageLayout } from "@/components/ui/layout/page-layout";

export const metadata: Metadata = {
	title: "Datatattat",
	description: "Generated by create next app",
};

export default async function RootLayout({
	children,
	params,
}: Readonly<{
	children: React.ReactNode;
	params: Promise<{ route: string[] }>;
}>) {
	const { route } = await params;

	return (
		<PageLayout>
			<MainHeader route={route} />
			{children}
		</PageLayout>
	);
}
