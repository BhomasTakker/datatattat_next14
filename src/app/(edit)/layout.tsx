import { MainHeader } from "@/components/header/main-header";
import { PageLayout } from "@/components/ui/layout/page-layout";
import { ToastContainer } from "@/lib/sonner/toast-container";

export default async function RootLayout({
	children,
}: Readonly<{
	children: React.ReactNode;
	params: Promise<{ route: string[] }>;
}>) {
	return (
		<PageLayout>
			<MainHeader />
			{children}
			<ToastContainer />
		</PageLayout>
	);
}
