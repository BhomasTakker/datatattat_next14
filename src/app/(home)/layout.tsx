import { MainHeader } from "@/components/header/main-header";
import { PageLayout } from "@/components/ui/layout/page-layout";

export default async function RootLayout({
	children,
}: Readonly<{
	children: React.ReactNode;
}>) {
	return (
		<PageLayout>
			<MainHeader />
			{children}
		</PageLayout>
	);
}
