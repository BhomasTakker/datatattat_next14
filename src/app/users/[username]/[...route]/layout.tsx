import { PageLayout } from "@/components/ui/layout/page-layout";
import { MainHeader } from "@/components/header/main-header";

export default async function RootLayout({
	children,
	params,
}: Readonly<{
	children: React.ReactNode;
	params: Promise<{ username: string; route: string[] }>;
}>) {
	const { username, route } = await params;
	const headerRoute = ["users", username, ...route];

	return (
		<PageLayout>
			<MainHeader route={headerRoute} />
			{children}
		</PageLayout>
	);
}
