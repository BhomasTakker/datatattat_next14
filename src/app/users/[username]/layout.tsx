import type { Metadata } from "next";
import { PageLayout } from "@/components/ui/layout/page-layout";
import { MainHeader } from "@/components/header/main-header";

export const metadata: Metadata = {
	title: "Datatattat",
	description: "Generated by create next app",
};

export default async function RootLayout({
	children,
	params,
}: Readonly<{
	children: React.ReactNode;
	params: Promise<{ username: string }>;
}>) {
	const { username } = await params;
	const route = ["users", username];

	return (
		<PageLayout>
			<MainHeader route={route} />
			{children}
		</PageLayout>
	);
}
