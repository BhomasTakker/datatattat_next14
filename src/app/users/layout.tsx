import { MainHeader } from "@/components/header/main-header";
import { PageLayout } from "@/components/ui/layout/page-layout";

export default async function RootLayout({
	children,
	params,
}: Readonly<{
	children: React.ReactNode;
	params: Promise<{ route: string[] }>;
}>) {
	const { route } = await params;

	return (
		<PageLayout>
			<MainHeader route={route} />
			{children}
		</PageLayout>
	);
}
