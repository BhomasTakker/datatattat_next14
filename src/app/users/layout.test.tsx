import "@testing-library/jest-dom";
import { render, screen } from "@testing-library/react";
import RootLayout from "./layout";
import { PageLayout } from "@/components/ui/layout/page-layout";

jest.mock("../../components/ui/layout/page-layout", () => ({
	PageLayout: jest.fn(({ children }: any) => (
		<div data-testid="page-display">{children}</div>
	)),
}));

describe("RootLayout", () => {
	afterEach(() => {
		jest.clearAllMocks();
	});
	it("renders children correctly", async () => {
		// @ts-expect-error - children type JSX or ReactNode
		render(await RootLayout({ children: <span>Test Child</span> }));
		expect(PageLayout).toHaveBeenCalledTimes(1);
		expect(screen.getByText("Test Child")).toBeInTheDocument();
	});

	it("calls PageLayout Once", async () => {
		// @ts-expect-error - children type JSX or ReactNode
		render(await RootLayout({ children: <span>Test Child</span> }));
		expect(PageLayout).toHaveBeenCalledTimes(1);
	});

	describe("Snaps", () => {
		it("renders Layout Unchanged", async () => {
			const { container } = render(
				// @ts-expect-error - children type JSX or ReactNode
				await RootLayout({ children: <span>Test Child</span> })
			);
			expect(container).toMatchSnapshot();
		});
	});
});
