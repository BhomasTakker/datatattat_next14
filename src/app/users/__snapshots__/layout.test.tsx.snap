// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RootLayout Snaps renders Layout Unchanged 1`] = `
<div>
  <div
    data-testid="page-display"
  >
    <span>
      Test Child
    </span>
  </div>
</div>
`;
